docs/blog/2020-01-30-why-gatsby-is-better-with-javascript/index.md

```diff
@@ -0,0 +1,60 @@
+
+Gatsby isn’t the only framework that produces a Single-Page App. This is a problem for a big part of the web at the moment. We can’t solve it alone or on a user level - there should be more focused efforts to establish accessibility standards and initiatives that address issues that are unique to web apps. Without those efforts a big part of web will always be less accessible.
+
+### Performance
+
+While using JavaScript in Gatsby can offer performance benefits, it comes with the usual caveat — if you use too much, performance will suffer. It's possible to have too much JavaScript, both in page data files and actual code bundles. The former can happen if too much data is requested for the page or put in the page context (some bad pagination patterns can cause that). With JavaScript, adding too many third-party libraries can easily cause that. It's important to monitor how much data you include on the page and in the future we plan to warn about large bundles during Gatsby compilation. It’s also possible to change the React runtime to alternative smaller runtimes like Preact by using a [Preact plugin](https://www.gatsbyjs.org/packages/gatsby-plugin-preact/).
+
+Another approach to make JavaScript bundles more optimal is [“differential building”](https://github.com/gatsbyjs/gatsby/issues/2114). Future Gatsby updates will produce multiple bundles, based on different versions of the platform that browsers support. Modern ones support more APIs and thus they don’t require as many shims or polyfills. A CDN can detect which browser a visitor is using and select an appropriate bundle to send.
+
+### Partial Hydration
+
+In general, downloading and parsing JavaScript, and doing [hydration](https://www.gatsbyjs.org/docs/glossary/#hydration) is still a cost that every Gatsby website has to pay. In particular the rehydration for big pages can be lengthy, while not being strictly required for the interactivity. Ideally Gatsby would make only critical parts interactive first and then rehydrate the rest of the page asynchronously. In the future, that should be possible because React will have [partial hydration](https://github.com/facebook/react/issues/13206). Gatsby will let developers specify critical parts of their application that need to be hydrated. It should also be possible to use analytics to learn which components on a page are critical based on user interactions, and automatically hydrate critical elements first.
```

Comment:
```suggestion
In general, downloading and parsing JavaScript, and doing [hydration](https://www.gatsbyjs.org/docs/glossary/#hydration) is still a cost that every Gatsby website has to pay. In particular, the rehydration for big pages can be lengthy, while not being strictly required for the interactivity. Ideally, Gatsby would make only critical parts interactive first and then rehydrate the rest of the page asynchronously. In the future, that should be possible because React will have [partial hydration](https://github.com/facebook/react/issues/13206). Gatsby will let developers specify critical parts of their application that need to be hydrated. It should also be possible to use analytics to learn which components on a page are critical based on user interactions and automatically hydrate critical elements first.
```